File: placementprepration/maxsliding.java


public class maxsliding{
    public static int[] maxsli(int[] arr,int k){
        int n=arr.length;
        int [] result=new int[n-k+1];
        for(int i=0;i<n-k;i++){
            int max=arr[i];
            for(int j=i;j<i+k;j++){
                max=Math.max(max, arr[j]);
            }
            result[i]=max;
        }
        return result;

    }
}

File: placementprepration/maxsliding.py
import sys


def max_sliding(arr, k):
    n = len(arr)
    if k == 0 or k > n:
        return []
    res = []
    for i in range(n - k + 1):
        res.append(max(arr[i : i + k]))
    return res


def main():
    data = list(map(int, sys.stdin.read().strip().split()))
    if len(data) < 2:
        return
    n, k, *rest = data
    arr = rest[:n]
    res = max_sliding(arr, k)
    print(" ".join(map(str, res)))


if __name__ == "__main__":
    main()
